### ISSUE TITLE
Exception thrown when 3rd party class library attempts to inspect property SqlContext.IsAvailable

### ISSUE THREAD

**Issue by jonathanpickard** [dotnet/corefx#30733]
Hello, I'm using a 3rd party class library that up to a certain point, works OK with .Net Core 2.1.1 and the Microsoft.Windows.Compatibility library. However, this 3rd party class library is throwing an exception because it is attempting to do this: if (SqlContext.IsAvailable)
The exception thrown is: Could not load type 'Microsoft.SqlServer.Server.SqlContext' from assembly 'System.Data
Please could you let me know if .Net Core or the Windows Compatibility Library will ever support the Microsoft.SqlServer.Server.SqlContext namespace? If not then is there any workaround? If there is no workaround then should I fall back on to .Net Framework 4.7.2 for my API?

Here is the most relevant part of the exception stack:
 {System.TypeInitializationException: The type initializer for 'PayrollEngine.GrossToNet' threw an exception. ---> System.TypeInitializationException: The type initializer for 'Fujitsu.COBOL.Runtime.ProgramControl.Management' threw an exception. ---> System.TypeLoadException: Could not load type 'Microsoft.SqlServer.Server.SqlContext' from assembly 'System.Data, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089'.
   at Fujitsu.COBOL.Runtime.ProgramControl.PCB..ctor()
   at Fujitsu.COBOL.Runtime.ProgramControl.Management..cctor()
   --- End of inner exception stack trace ---
   at Fujitsu.COBOL.Runtime.ProgramControl.LIA..ctor(Byte[] zaoption, Object[] zacnttb, String[] zaprgnm, String zaclsnm, Type zatype, CodeRegion zacodeRegion)
   at Fujitsu.COBOL.Runtime.ProgramControl.Management.JMP1PLAI(Byte[] zaoption, Object[] zacnttb, String[] zaprgnm, String zaclsnm, Type zatype)
   at PayrollEngine.GrossToNet..cctor() in C:\PATesting\12.1\Source\Payroll Engine for API\Gross2Net\CobolGrossToNet\Source Files\ZX21UW.COB:line 0

**Comment by keeratsingh** [comment]
Hi @jonathanpickard,
>Please could you let me know if .Net Core or the Windows Compatibility Library will ever support the Microsoft.SqlServer.Server.SqlContext namespace? If not then is there any workaround? If there is no workaround then should I fall back on to .Net Framework 4.7.2 for my API?

To give more context currently `Microsoft.SqlServer.Server.SqlContext`  APIs are not yet ported to .NET Core, even if `Microsoft.SqlServer.Server.SqlContext` APIs were imported to .NET Core it still wouldn’t guarantee the .NET Core code running in SQL Server, as .NET Core code is not supported inside SQL Server.

Also, as per https://github.com/dotnet/corefx/issues/23292#issuecomment-402709831, you can try adding NuGet package `Microsoft.SqlServer.SqlManagementObjects` as it doesn't reference SqlContext.

>The NuGet package Microsoft.SqlServer.SqlManagementObjects has versions of the SMO binaries that target NetStandard 2.0

If the reply answers you questions I will close this issue as GitHub Issue #23292 already exists that tracks this issue.

**Comment by keeratsingh** [comment]
Closing issue as there was no further response by the user.

**Comment by jonathanpickard** [comment]
Hello,

I've upgraded to VS 2019 Preview 3 and the latest .Net Core Framework Preview.  I've also added the latest SqlManagementObjects package but the error exception message hasn't been fixed.  I've even added the latest SQL client library.  Here are all of the package references in my project, followed by the error message:

Id                                  Versions                                 ProjectName                                  
--                                  --------                                 -----------                                  
Microsoft.EntityFrameworkCore       {3.0.0-preview.19074.3}                  
Microsoft.Windows.Compatibility     {2.0.1}                                  
Microsoft.NETCore.Platforms         {3.0.0-preview.19073.11}                 
Microsoft.AspNetCore.App            {3.0.0-preview-19075-0444}               
Microsoft.SqlServer.SqlManagemen... {150.18085.0-preview}                    
Swashbuckle.AspNetCore              {4.0.1}                                  
Microsoft.NETCore.App               {3.0.0-preview-27324-5}                  
NETStandard.Library                 {2.0.3}                                  
Microsoft.EntityFrameworkCore.In... {3.0.0-preview.19074.3}                  
System.Data.SqlClient               {4.7.0-preview.19073.11}                 
runtime.win-x64.Microsoft.NETCor... {3.0.0-preview-27324-5}                  


TypeLoadException: Could not load type 'Microsoft.SqlServer.Server.SqlContext' from assembly 'System.Data, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089'.


---

If you've got any more ideas then I'd be happy to try them.
